New cmake build system removed addFunction support

I was export addFunction using make.py

```
  if add_function_support:
    args += ' -s RESERVED_FUNCTION_POINTERS=20 -s EXTRA_EXPORTED_RUNTIME_METHODS=["addFunction"]'
```

So i can do stuff like
```
                    const contactAddedCallbackPtr = Ammo.addFunction((cp:any, colObj0Wrap:any, partId0:number, index0:number, colObj1Wrap, partId1:number, index1:number) => {
                        // NOTE: KEEP FOR REFERENCE
                        // colObj0Wrap = Ammo.wrapPointer(colObj0Wrap, Ammo.btCollisionObjectWrapper);
                        // colObj1Wrap = Ammo.wrapPointer(colObj1Wrap, Ammo.btCollisionObjectWrapper);
                        // const colobj0:any = colObj0Wrap.getCollisionObject();
                        // const colobj1:any = colObj1Wrap.getCollisionObject();
                        // if (colobj0 != null && colobj1 != null) {
                        //    const colshape0:any = colobj0.getCollisionShape();
                        //    const colshape1:any = colobj1.getCollisionShape();
                        // }
                        if (physicsplugin.world.adjustInternalEdgeContacts) {
                            physicsplugin.world.adjustInternalEdgeContacts(cp, colObj1Wrap, colObj0Wrap, partId1, index1);
                            return true;
                        } else {
                            return false;
                        }
                    });
```


Now with the new cmake -B builds ... it does not support 

```
$ cmake -B builds -DRESERVED_FUNCTION_POINTERS=20 -DEXTRA_EXPORTED_RUNTIME_METHODS=["addFunction"] -DTOTAL_MEMORY=134217728

```

So how am i supposed to get Ammo.addFunction back ???

Could it be something missing from my machine ???

Note: I tried the modded make.py

but got different error:

```
make: *** [config.status] Error 77
Traceback (most recent call last):
  File "make.py", line 194, in <module>
    build()
  File "make.py", line 159, in build
    emscripten.make(['make', '-j', str(CORES)])
  File "/users/mackey/tools/emsdk/upstream/emscripten/tools/building.py", line 382, in make
    run_process(args, stdout=stdout, stderr=stderr, env=env, shell=WINDOWS, **kwargs)
  File "/users/mackey/tools/emsdk/upstream/emscripten/tools/shared.py", line 203, in run_process
    result.check_returncode()
  File "/users/mackey/tools/emsdk/upstream/emscripten/tools/shared.py", line 176, in check_returncode
    raise Py2CalledProcessError(returncode=self.returncode, cmd=self.args, output=self.stdout, stderr=self.stderr)
tools.shared.Py2CalledProcessError: Command '['make', '-j', '24']' returned non-zero exit status 2

```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -103,7 +103,7 @@ add_custom_target(ammo-bindings ALL DEPENDS glue.js glue.o)
 add_custom_command(
   OUTPUT ammo.js
   COMMAND emcc glue.o ${BULLET_LIBS} ${EMCC_JS_ARGS} -o ammo.js
-  DEPENDS ammo-bindings ${BULLET_TARGETS}
+  DEPENDS ammo-bindings ${AMMO_FRONT_MATTER_FILE} ${AMMO_ONLOAD_FILE} ${BULLET_TARGETS}
   COMMENT "Building ammo javascript"
   VERBATIM)
 add_custom_target(ammo-javascript ALL DEPENDS ammo.js)
@@ -113,7 +113,7 @@ add_custom_target(ammo-javascript ALL DEPENDS ammo.js)
 add_custom_command(
   OUTPUT ammo.wasm.js ammo.wasm.wasm
   COMMAND emcc glue.o ${BULLET_LIBS} ${EMCC_WASM_ARGS} -o ammo.wasm.js
-  DEPENDS ammo-bindings ${BULLET_TARGETS}
+  DEPENDS ammo-bindings ${AMMO_FRONT_MATTER_FILE} ${AMMO_ONLOAD_FILE} ${BULLET_TARGETS}
   COMMENT "Building ammo webassembly"
   VERBATIM)
 add_custom_target(ammo-wasm ALL DEPENDS ammo.wasm.js ammo.wasm.wasm)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,6 +6,7 @@ set(WEBIDL_BINDER_SCRIPT $ENV{EMSCRIPTEN}/tools/webidl_binder.py)
 set(AMMO_FRONT_MATTER_FILE ${CMAKE_CURRENT_SOURCE_DIR}/front-matter.js)
 set(AMMO_HEADER_FILE ${CMAKE_CURRENT_SOURCE_DIR}/ammo.h)
 set(AMMO_IDL_FILE ${CMAKE_CURRENT_SOURCE_DIR}/ammo.idl)
+set(AMMO_ONLOAD_FILE ${CMAKE_CURRENT_SOURCE_DIR}/onload.js)
 set(BULLET_SRC_DIR ${CMAKE_CURRENT_SOURCE_DIR}/bullet/src/)
 
 set(BULLET_TARGETS BulletCollision BulletDynamics BulletSoftBody LinearMath)
@@ -32,6 +33,7 @@ set(EMCC_ARGS
   --extern-pre-js ${AMMO_FRONT_MATTER_FILE}
   --llvm-lto 1
   --post-js glue.js
+  --post-js ${AMMO_ONLOAD_FILE}
   -O3
   -s ALLOW_MEMORY_GROWTH=${ALLOW_MEMORY_GROWTH}
   -s EXPORTED_RUNTIME_METHODS=["UTF8ToString"]

--- a/onload.js
+++ b/onload.js
@@ -1,0 +1,2 @@
+// Reassign global Ammo to the loaded module:
+this['Ammo'] = Module;

--- a/tests/loading.js
+++ b/tests/loading.js
@@ -1,0 +1,7 @@
+const test = require('ava');
+const loadAmmo = require('./helpers/load-ammo.js');
+
+test('loading should assign global Ammo', async t => {
+  await loadAmmo()
+  t.assert(Ammo)
+})

